backend/compact-connect/lambdas/staff-users/tests/function/test_data_model/test_client.py

```diff
@@ -0,0 +1,442 @@
+
+        # Verify we're seeing the expected sorting
+        family_names = [user['attributes']['familyName'] for user in resp['items']]
+        sorted_family_names = sorted(family_names)
+        self.assertEqual(sorted_family_names, family_names)
+
+    def test_get_jurisdictions_users_by_family_name(self):
+        # One user with compact-staff-like permissions in aslp
+        self._create_compact_staff_user(compacts=['aslp'])
+        # One user with board-staff-like permissions in aslp in each jurisdiction
+        self._create_board_staff_users(compacts=['aslp'])
+        # One user with board-staff-like permissions in aslp and octp in each jurisdiction
+        self._create_board_staff_users(compacts=['aslp', 'octp'])
+
+        from data_model.client import UserClient
+
+        client = UserClient(self.config)
+
+        # Provide a client filter that will filter out users without permissions in the jurisdiction we're looking for
+        resp = client.get_users_sorted_by_family_name(  # pylint: disable=missing-kwoa
+            compact='aslp',
+            # All three jurisdictions, this time
+            jurisdictions=['oh', 'ne', 'ky']
+        )
+
+        # We created two board users that have aslp permissions in each jurisdiction so those are what we should get
+        # back
+        self.assertEqual(2*len(self.config.jurisdictions), len(resp['items']))
+
+        # Verify that we're getting the expected fields
+        for user in resp['items']:
+            self.assertEqual(
+                {'type', 'userId', 'compact', 'attributes', 'permissions', 'dateOfUpdate'},
+                user.keys()
+            )
+
+        # Verify we're seeing the expected sorting
+        family_names = [user['attributes']['familyName'] for user in resp['items']]
+        sorted_family_names = sorted(family_names)
+        self.assertEqual(sorted_family_names, family_names)
+
+    def test_get_one_jurisdiction_users_by_family_name(self):
+        # One user with compact-staff-like permissions in aslp
+        self._create_compact_staff_user(compacts=['aslp'])
+        # One user with board-staff-like permissions in aslp in each jurisdiction
+        self._create_board_staff_users(compacts=['aslp'])
+        # One user with board-staff-like permissions in aslp and octp in each jurisdiction
+        self._create_board_staff_users(compacts=['aslp', 'octp'])
+
+        from data_model.client import UserClient
+
+        client = UserClient(self.config)
+
+        # Provide a client filter that will filter out users without permissions in the jurisdiction we're looking for
+        resp = client.get_users_sorted_by_family_name(  # pylint: disable=missing-kwoa
+            compact='aslp',
+            # All three jurisdictions, this time
+            jurisdictions=['oh']
+        )
+
+        # We created two board users that have aslp permissions in oh so those are what we should get back
+        self.assertEqual(2, len(resp['items']))
```

Comment:
Why is this just 2 vs line 94, which is 2*the number of jurisdictions?

Reply:
In this case, we are filtering the results down to a single jurisdiction (`oh`) but in line 94, we are looking at each of three different jurisdictions (there are only three jurisdictions (`oh`, `ne`, `ky`) in this test suite, to keep size reasonable).

Reply:
ah got it. so still that same calculation, just putting the number down explicitly